scio-test/src/test/scala/com/spotify/scio/testing/SCollectionMatchersTest.scala

```diff
@@ -133,6 +133,48 @@ class SCollectionMatchersTest extends PipelineSpec {
     }
   }
 
+  it should "support beCloseTo" in {
+    // should cases
+    runWithContext {
+      _.parallelize(Seq(9, 10, 11)) should beCloseTo[Int](9, 3)
```

Comment:
This feels more like `should beAllCloseTo`.
I guess there's no way to test the case where both `actual` and `expected` are collections?

Also is the follower a more readable alternative to `beCloseTo`?

```scala
sc.parallelize(Seq(8.9, 9.0, 9.1)) should forAll[Double] {
  _ === 9.0 +- 0.1 // Double => Boolean
}
```

This would work with `satisfy`, `forAll`, `exists`, etc. Right now it fails serialization but should be fixable?

```
[info]   Cause: java.io.NotSerializableException: org.scalatest.Assertions$AssertionsHelper
[info]  - field (class "org.scalatest.FlatSpec", name: "assertionsHelper", type: "class org.scalatest.Assertions$AssertionsHelper")
[info]  - object (class "com.spotify.scio.testing.SCollectionMatchersTest", SCollectionMatchersTest)
[info]  - field (class "com.spotify.scio.testing.SCollectionMatchersTest$$anonfun$7", name: "$outer", type: "class com.spotify.scio.testing.SCollectionMatchersTest")
[info]  - object (class "com.spotify.scio.testing.SCollectionMatchersTest$$anonfun$7", <function0>)
[info]  - field (class "com.spotify.scio.testing.SCollectionMatchersTest$$anonfun$7$$anonfun$apply$90", name: "$outer", type: "class com.spotify.scio.testing.SCollectionMatchersTest$$anonfun$7")
[info]  - object (class "com.spotify.scio.testing.SCollectionMatchersTest$$anonfun$7$$anonfun$apply$90", <function1>)
[info]  - field (class "com.spotify.scio.testing.SCollectionMatchersTest$$anonfun$7$$anonfun$apply$90$$anonfun$apply$2", name: "$outer", type: "class com.spotify.scio.testing.SCollectionMatchersTest$$anonfun$7$$anonfun$apply$90")
[info]  - object (class "com.spotify.scio.testing.SCollectionMatchersTest$$anonfun$7$$anonfun$apply$90$$anonfun$apply$2", <function1>)
[info]  - field (class "com.spotify.scio.testing.SCollectionMatchers$$anonfun$forAll$1", name: "predicate$2", type: "interface scala.Function1")
[info]  - object (class "com.spotify.scio.testing.SCollectionMatchers$$anonfun$forAll$1", <function1>)
[info]  - field (class "com.spotify.scio.testing.SCollectionMatchers$$anon$6$$anon$13$$anonfun$5", name: "p$1", type: "interface scala.Function1")
[info]  - object (class "com.spotify.scio.testing.SCollectionMatchers$$anon$6$$anon$13$$anonfun$5", <function1>)
[info]  - field (class "com.spotify.scio.testing.SCollectionMatchers$$anon$7", name: "f$1", type: "interface scala.Function1")
[info]  - object (class "com.spotify.scio.testing.SCollectionMatchers$$anon$7", com.spotify.scio.testing.SCollectionMatchers$$anon$7@5d97d758)
[info]  - field (class "org.apache.beam.sdk.testing.PAssert$GroupedValuesCheckerDoFn", name: "checkerFn", type: "interface org.apache.beam.sdk.transforms.SerializableFunction")
[info]  - root object (class "org.apache.beam.sdk.testing.PAssert$GroupedValuesCheckerDoFn", org.apache.beam.sdk.testing.PAssert$GroupedValuesCheckerDoFn@7cda0d26)
```